intera_interface/src/intera_motion_interface/motion_trajectory.py

```diff
@@ -170,3 +171,31 @@ def to_csv_file(self, file_name):
         with open(file_name, "wb") as f:
             writer = csv.writer(f)
             writer.writerows(waypoint_data)
+
+    def _check_options(self):
+        """
+        Check to make sure the defined waypoints have the necessary parameters
+        given the provided options and correct the waypoints when possible.
+        """
+        if (self._traj.trajectory_options.interpolation_type ==
+            TrajectoryOptions.CARTESIAN):
+            self._check_cartesian_options()
+
+    def _check_cartesian_options(self):
+        """
+        Checks the waypoints to make sure that cartesian endpoints are defined
+        and attempts to calculate them if possible
+        """
+        for index in range(len(self._traj.waypoints)):
+            wpt = self._traj.waypoints[index]
```

Comment:
There is probably a better python way of looping through the waypoints directly without using an index.